### DISCUSSION TITLE
Glb with Animation

### DISCUSSION THREAD

**Discussion by hvo2004** [sceneview/sceneview#65] [Q&A]
Can an ArModelNode animate? I loaded https://sceneview.github.io/assets/models/Halloween.glb (has 1 animation) with autoAnimate = true; however the model does not animate at all...

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, GLB animations play automatically when loaded. You can control them via the `ModelInstance`:

```kotlin
val model = rememberModelInstance(modelLoader, "models/animated_model.glb")
if (model != null) {
    // Animations from the GLB are available on the model's animator
    val animator = model.animator
    // Play first animation
    animator?.apply {
        playAnimation(0)
    }
    ModelNode(modelInstance = model)
}
```

Animations embedded in GLB/glTF files (skeletal, morph targets, transform) are all supported through the Filament animator.
